util/build_configs/cray-internal/setenv-xc-aarch64.bash

```diff
@@ -0,0 +1,374 @@
+        echo "    -B COMPONENT : Chapel component to build (optional, repeatable)"
+        echo "               Valid components: '$valid_components'"
+        echo "          Default if -B omitted: '$default_components'"
+        echo "          Selected components are always built in the order shown above."
+        echo "          COMPONENT may be prefixed with a plus sign or a minus sign."
+        echo "          For example, if '-B +clean' is given (as the only -B parameter),"
+        echo "          then the default list of components will be built, followed by"
+        echo "          'make cleanall'. If '-B clean' was given instead, then"
+        echo "          'make cleanall' would be the ONLY thing done by this script."
+        exit 1
+    }
+    while getopts :vnB:h opt; do
+        case $opt in
+        ( v ) verbose=-v ;;
+        ( n ) dry_run=-n ;;
+        ( B )
+            # ck valid component
+            case ",${OPTARG#[+-]}," in
+            ( *,compiler,* | *,runtime,* | *,venv,* | *,mason,* | *,clean,* ) ;;
+            ( * )   log_error "Invalid -B COMPONENT='$OPTARG'"; usage ;;
+            esac
+            # if + or - prefix, modify the current list of selected components
+            case "$OPTARG" in
+            ( +* )  components="${components:-$default_components}" ;;
+            ( -* )  components= ;;
+            esac
+            # add option to current list of selected components
+            components="${components:+$components,}$OPTARG"
+            ;;
+        ( h ) usage;;
+        ( \?) log_error "Invalid option: -$OPTARG"; usage;;
+        ( : ) log_error "Option -$OPTARG requires an argument."; usage;;
+        esac
+    done
+
+    log_info "Begin project $setenv"
+
+    export CHPL_HOME=${CHPL_HOME:-$( cd $cwd/../../.. && pwd )}
+    log_debug "with CHPL_HOME=$CHPL_HOME"
+    ck_chpl_home "$CHPL_HOME"
+
+    # if no components were given on cmdline, then use the default list
+    if [ -z "$components" ]; then
+        components=$default_components
+    fi
+    log_info "Selected Chapel components: $components"
+    log_info "Valid Chapel components:    $valid_components"
+
+    # Default config
+
+##  # If the CHPL_ env variable is NOT defined here (and not overridden
+##  # by build_configs.py cmdline arguments), Chapel make will determine
+##  # the appropriate default value for the platform / environment it finds.
+
+    export CHPL_HOST_PLATFORM=cray-xc
+    export CHPL_TARGET_PLATFORM=cray-xc
+    export CHPL_REGEXP=re2      # re2 required for mason
+    export CHPL_LOCAL_MODEL=flat
+    export CHPL_COMM=none
+    export CHPL_COMM_SUBSTRATE=none
+    export CHPL_TASKS=qthreads
+    export CHPL_LAUNCHER=none
+    export CHPL_LLVM=llvm       # llvm requires cmake
+    export CHPL_AUX_FILESYS=none
+
+    # More CPUs --> faster make. Or, unset CHPL_MAKE_MAX_CPU_COUNT to use all CPUs.
+
+    export CHPL_MAKE_MAX_CPU_COUNT=${CHPL_MAKE_MAX_CPU_COUNT:-24}
```

Comment:
I still think 24 is too large a default, by a long shot.  Cray X* systems are always shared by many users, and defaulting to use all the cores on a login node is rude.  Also, it isn't necessarily "all" CPUs as the comment claims; the aarch64 login nodes actually have quite a bit than 24 cores.

Reply:
Addressed